lib/api/core/indexCache.js

```diff
@@ -48,58 +49,43 @@ class IndexCache {
   }
 
   init () {
-    const engine = this.kuzzle.internalEngine;
-    const kuzzleInfoFieldsRegex = new RegExp(`^_kuzzle_info\\.(${Object.keys(this.commonMapping._kuzzle_info.properties).join('|')})$`);
+    const internalEngine = this.kuzzle.internalEngine;
 
-    return engine.listIndexes()
+    return internalEngine.listIndexes()
       .then(indexes => indexes.filter(index => index !== this.kuzzle.internalEngine.index))
       .then(indexes => {
         const promises = [];
 
-        for (const index of indexes.filter(idx => idx !== this.kuzzle.internalEngine.index)) {
+        for (const index of indexes) {
           this.indexes[index] = [];
           this.defaultMappings[index] = _.cloneDeep(this.commonMapping);
 
-          promises.push(engine.getFieldMapping({
-            index,
-            fields: '_kuzzle_info.*'
-          })
-            .then(kuzInfoMappings => {
-              if (kuzInfoMappings[index] && kuzInfoMappings[index].mappings) {
-                for (const collection of Object.keys(kuzInfoMappings[index].mappings)) {
-                  const collectionInfo = kuzInfoMappings[index].mappings[collection];
-
-                  for (const dotField of Object.keys(collectionInfo).filter(f => kuzzleInfoFieldsRegex.test(f))) {
-                    Object.assign(this.defaultMappings[index]._kuzzle_info.properties, collectionInfo[dotField].mapping);
-                  }
-                  // do not break to catch cases where not all fields are defined
+          promises.push(
+            internalEngine.listCollections(index)
+              .then(collections => {
+                for (const collection of collections) {
+                  promises.push(
```

Comment:
Somehow, I don't think this is a good idea, because depending on how bluebird coded their `Promise.all` feature, it might or might not work.
A promises array, once it has been populated, should not be muted.

I think that it would be better, to create a new promise array and, at the end of that promise subfunction, to return `Bluebird.all(<that new promises array>)` instead of null.

Another benefit to that is that you could flatten that promise pyramid a bit, because `.all` returns an array of promises results, that you could process in one pass instead of having 1 then() applied to each applyDefaultMapping promise result